src/web/pages.h

```diff
@@ -137,8 +137,8 @@ class AddObject : public WebRequestHandler {
 
 protected:
     void addContainer(int parentID);
-    std::shared_ptr<CdsObject> addItem(int parentID, const std::shared_ptr<CdsItem>& item);
-    std::shared_ptr<CdsObject> addUrl(int parentID, const std::shared_ptr<CdsItemExternalURL>& item, bool addProtocol);
+    std::shared_ptr<CdsObject> addItem(int parentID);
+    std::shared_ptr<CdsObject> addUrl(int parentID, bool addProtocol);
```

Comment:
What about changing the return to the sub class, i.e CdsItem and CdsExternalItem